packages/amplify-appsync-simulator/src/data-loader/dynamo-db/index.ts

```diff
@@ -60,6 +63,37 @@ export class DynamoDBDataLoader implements AmplifyAppSyncSimulatorDataLoader {
     }
   }
 
+  private async deleteAllItems(): Promise<object | null> {
+    try {
+      const items = await this.getAllItems();
+      for await (const item of items) {
+        await this.client
+          .deleteItem({
+            TableName: this.tableName,
+            Key: { id: item.id },
+            ReturnValues: 'ALL_OLD',
+          }).promise();
+      }
+    } catch (e) {
+      throw new Error(`Error while deleting all items from ${this.tableName}`);
+    }
+    return [this.tableName];
+  }
+
+  private async getAllItems(): Promise<Array<AttributeMap> | null> {
```

Comment:
Similarly, a comment on this method wouldn't go amiss.

Reply:
As I read this method, kind of a silly thought that may be simpler.

Can you just continually run a scan on page 1, verify you have items.length > 0 && LastEvaluatedKey !== undefined, then call BatchWriteItems w/ deletes for those items, and break when you get no records back? Then you don't need to worry about holding a ton of items in-memory and passing them between these methods.